api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -123,6 +123,20 @@ message CommonTlsContext {
     }
   }
 
+  // Similar to CertificateProvider above, but allows the plugin instances to be configured on the
+  // client side instead of being sent from the control plane.
+  message CertificateProviderInstance {
+    // Plugin instance name. This name must be defined in the client's configuration (e.g., a
+    // bootstrap file) to correspond to a plugin instance (i.e., the same data in the typed_config
+    // field that would be sent in the CertificateProvider message if the config was sent by the
+    // control plane). If not present, defaults to "default".
+    string plugin_instance_name = 1;
```

Comment:
Previously someone commented on the use of the word "plugin" in this context - IIRC the preferred word is "extension". Will need to be changed in the name of the field and the comments. But I am okay with this word.